From 21af6234c106eaa4aa17f1e15e965fc6761439c3 Mon Sep 17 00:00:00 2001
From: Abhay Singh Gill <abhaygill017@gmail.com>
Date: Tue, 24 Dec 2024 11:08:26 +0530
Subject: [PATCH 3/4] Launcher3: Update grid sizes from Pixel launcher

* This is a minimal import from Pixel launcher
* Adapt 5+ column grids to be scalable as well
* Remove 6 column grids for phones as well those are only meant for tablets.

Change-Id: I233e9f39b2673d3bd2dbb926de9b2445b23c39e2
Signed-off-by: Abhay Singh Gill <abhaygill017@gmail.com>
---
 res/values-land/styles.xml                   |  20 +-
 res/values-sw600dp-land/styles.xml           |  10 +
 res/values-sw640dp-land/styles.xml           |  14 +
 res/values-sw640dp/styles.xml                |  14 +
 res/values-sw700dp/styles.xml                |   8 +
 res/values/styles.xml                        |  50 ++
 res/xml/default_workspace_6x6.xml            | 171 ------
 res/xml/device_profiles.xml                  | 519 +++++++++----------
 src/com/android/launcher3/LauncherFiles.java |   2 -
 9 files changed, 366 insertions(+), 442 deletions(-)
 create mode 100644 res/values-sw600dp-land/styles.xml
 create mode 100644 res/values-sw640dp-land/styles.xml
 create mode 100644 res/values-sw640dp/styles.xml
 create mode 100644 res/values-sw700dp/styles.xml
 delete mode 100644 res/xml/default_workspace_6x6.xml

diff --git a/res/values-land/styles.xml b/res/values-land/styles.xml
index 52474dac52..2d447cd0d5 100644
--- a/res/values-land/styles.xml
+++ b/res/values-land/styles.xml
@@ -15,7 +15,25 @@
   -->
 
 <resources>
+    <style name="AllAppsStyle6x5" parent="@style/AllAppsStyleDefault">
+        <item name="horizontalPadding">32dp</item>
+    </style>
+    <style name="CellStyle6x5">
+        <item name="iconDrawablePadding">8dp</item>
+    </style>
     <style name="CellStyleDefault">
         <item name="iconDrawablePadding">4dp</item>
     </style>
-</resources>
\ No newline at end of file
+    <style name="Folder6x5Style" parent="@style/FolderStyleDefault">
+        <item name="folderCellHeight">74dp</item>
+        <item name="folderCellWidth">74dp</item>
+        <item name="folderFooterHeight">64dp</item>
+    </style>
+    <style name="FolderStyleDefault">
+        <item name="folderTopPadding">20dp</item>
+        <item name="folderCellHeight">94dp</item>
+        <item name="folderCellWidth">80dp</item>
+        <item name="folderBorderSpace">16dp</item>
+        <item name="folderFooterHeight">56dp</item>
+    </style>
+</resources>
diff --git a/res/values-sw600dp-land/styles.xml b/res/values-sw600dp-land/styles.xml
new file mode 100644
index 0000000000..d97efb3f2b
--- /dev/null
+++ b/res/values-sw600dp-land/styles.xml
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <style name="FolderStyleDefault" parent="">
+        <item name="folderBorderSpace">16dp</item>
+        <item name="folderCellHeight">94dp</item>
+        <item name="folderCellWidth">80dp</item>
+        <item name="folderFooterHeight">56dp</item>
+        <item name="folderTopPadding">24dp</item>
+    </style>
+</resources>
diff --git a/res/values-sw640dp-land/styles.xml b/res/values-sw640dp-land/styles.xml
new file mode 100644
index 0000000000..bc8b8abdca
--- /dev/null
+++ b/res/values-sw640dp-land/styles.xml
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <style name="AllAppsStyle6x5" parent="@style/AllAppsStyleDefault">
+        <item name="horizontalPadding">32dp</item>
+    </style>
+    <style name="CellStyle6x5" parent="@style/CellStyleDefault">
+        <item name="iconDrawablePadding">10dp</item>
+    </style>
+    <style name="Folder6x5Style" parent="@style/FolderStyleDefault">
+        <item name="folderCellHeight">79dp</item>
+        <item name="folderCellWidth">79dp</item>
+        <item name="folderFooterHeight">64dp</item>
+    </style>
+</resources>
diff --git a/res/values-sw640dp/styles.xml b/res/values-sw640dp/styles.xml
new file mode 100644
index 0000000000..27958a9793
--- /dev/null
+++ b/res/values-sw640dp/styles.xml
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <style name="AllAppsStyle6x5" parent="@style/AllAppsStyleDefault">
+        <item name="horizontalPadding">28dp</item>
+    </style>
+    <style name="CellStyle6x5" parent="@style/CellStyleDefault">
+        <item name="iconDrawablePadding">14dp</item>
+    </style>
+    <style name="Folder6x5Style" parent="@style/FolderStyleDefault">
+        <item name="folderCellHeight">104dp</item>
+        <item name="folderCellWidth">102dp</item>
+        <item name="folderFooterHeight">64dp</item>
+    </style>
+</resources>
diff --git a/res/values-sw700dp/styles.xml b/res/values-sw700dp/styles.xml
new file mode 100644
index 0000000000..71e303346a
--- /dev/null
+++ b/res/values-sw700dp/styles.xml
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <style name="Folder6x5Style" parent="@style/FolderStyleDefault">
+        <item name="folderCellHeight">104dp</item>
+        <item name="folderCellWidth">102dp</item>
+        <item name="folderFooterHeight">64dp</item>
+    </style>
+</resources>
diff --git a/res/values/styles.xml b/res/values/styles.xml
index aaea321e1f..e26d0be184 100644
--- a/res/values/styles.xml
+++ b/res/values/styles.xml
@@ -499,6 +499,48 @@
         <item name="android:windowDrawsSystemBarBackgrounds">true</item>
     </style>
 
+    <style name="Folder2x2Style" parent="@style/FolderStyleDefault">
+        <item name="folderCellHeight">199dp</item>
+        <item name="folderCellWidth">176dp</item>
+    </style>
+
+    <style name="Folder3x3Style" parent="@style/FolderStyleDefault">
+        <item name="folderCellHeight">128dp</item>
+        <item name="folderCellWidth">112dp</item>
+        <item name="folderFooterHeight">64dp</item>
+    </style>
+
+    <style name="Folder4x4Style" parent="@style/FolderStyleDefault">
+        <item name="folderCellHeight">94dp</item>
+        <item name="folderCellWidth">80dp</item>
+    </style>
+
+    <style name="Folder4x5Style" parent="@style/FolderStyleDefault">
+        <item name="folderCellHeight">94dp</item>
+        <item name="folderCellWidth">80dp</item>
+    </style>
+
+    <style name="Folder5x5Style" parent="@style/FolderStyleDefault">
+        <item name="folderCellHeight">79dp</item>
+        <item name="folderCellWidth">61dp</item>
+    </style>
+
+    <style name="Folder5x6Style" parent="@style/FolderStyleDefault">
+        <item name="folderCellHeight">79dp</item>
+        <item name="folderCellWidth">61dp</item>
+    </style>
+
+    <style name="Folder5x7Style" parent="@style/FolderStyleDefault">
+        <item name="folderCellHeight">79dp</item>
+        <item name="folderCellWidth">61dp</item>
+    </style>
+
+    <style name="Folder6x5Style" parent="@style/FolderStyleDefault">
+        <item name="folderCellHeight">90dp</item>
+        <item name="folderCellWidth">77dp</item>
+        <item name="folderFooterHeight">64dp</item>
+    </style>
+
     <style name="FolderStyleDefault">
         <item name="folderTopPadding">24dp</item>
         <item name="folderCellHeight">94dp</item>
@@ -507,10 +549,18 @@
         <item name="folderFooterHeight">56dp</item>
     </style>
 
+    <style name="CellStyle6x5" parent="@style/CellStyleDefault">
+        <item name="iconDrawablePadding">14dp</item>
+    </style>
+
     <style name="CellStyleDefault">
         <item name="iconDrawablePadding">7dp</item>
     </style>
 
+    <style name="AllAppsStyle6x5" parent="@style/AllAppsStyleDefault">
+        <item name="horizontalPadding">32dp</item>
+    </style>
+
     <style name="AllAppsStyleDefault">
         <item name="horizontalPadding">16dp</item>
     </style>
diff --git a/res/xml/default_workspace_6x6.xml b/res/xml/default_workspace_6x6.xml
deleted file mode 100644
index e0bad1cd1b..0000000000
--- a/res/xml/default_workspace_6x6.xml
+++ /dev/null
@@ -1,171 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<!-- Copyright (C) 2009 The Android Open Source Project
-
-     Licensed under the Apache License, Version 2.0 (the "License");
-     you may not use this file except in compliance with the License.
-     You may obtain a copy of the License at
-
-          http://www.apache.org/licenses/LICENSE-2.0
-
-     Unless required by applicable law or agreed to in writing, software
-     distributed under the License is distributed on an "AS IS" BASIS,
-     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-     See the License for the specific language governing permissions and
-     limitations under the License.
--->
-
-<favorites xmlns:launcher="http://schemas.android.com/apk/res-auto/com.android.launcher3">
-
-    <!-- Hotseat (We use the screen as the position of the item in the hotseat) -->
-    <!-- Dialer, Messaging, [Maps/Music], Browser, Camera -->
-    <resolve
-        launcher:container="-101"
-        launcher:screen="0"
-        launcher:x="0"
-        launcher:y="0" >
-        <favorite launcher:uri="#Intent;action=android.intent.action.DIAL;end" />
-        <favorite launcher:uri="tel:123" />
-        <favorite launcher:uri="#Intent;action=android.intent.action.CALL_BUTTON;end" />
-    </resolve>
-
-    <resolve
-        launcher:container="-101"
-        launcher:screen="1"
-        launcher:x="1"
-        launcher:y="0" >
-        <favorite launcher:uri="#Intent;action=android.intent.action.MAIN;category=android.intent.category.APP_MESSAGING;end" />
-        <favorite launcher:uri="sms:" />
-        <favorite launcher:uri="smsto:" />
-        <favorite launcher:uri="mms:" />
-        <favorite launcher:uri="mmsto:" />
-    </resolve>
-
-    <resolve
-        launcher:container="-101"
-        launcher:screen="2"
-        launcher:x="2"
-        launcher:y="0" >
-        <favorite launcher:uri="#Intent;action=android.intent.action.MAIN;category=android.intent.category.APP_CONTACTS;end" />
-    </resolve>
-
-    <resolve
-        launcher:container="-101"
-        launcher:screen="3"
-        launcher:x="3"
-        launcher:y="0" >
-        <favorite launcher:uri="#Intent;action=android.intent.action.MAIN;category=android.intent.category.APP_BROWSER;end" />
-        <favorite launcher:uri="http://www.example.com/" />
-    </resolve>
-
-    <resolve
-        launcher:container="-101"
-        launcher:screen="4"
-        launcher:x="4"
-        launcher:y="0" >
-        <favorite launcher:uri="#Intent;action=android.media.action.STILL_IMAGE_CAMERA;end" />
-        <favorite launcher:uri="#Intent;action=android.intent.action.CAMERA_BUTTON;end" />
-    </resolve>
-
-    <!-- Screen 0 -->
-    <appwidget
-        launcher:packageName="com.google.android.googlequicksearchbox"
-        launcher:className="com.google.android.googlequicksearchbox.SearchWidgetProvider"
-        launcher:screen="0"
-        launcher:x="0"
-        launcher:y="0"
-        launcher:spanX="5"
-        launcher:spanY="1" />
-
-    <appwidget
-        launcher:packageName="com.android.deskclock"
-        launcher:className="com.android.alarmclock.DigitalAppWidgetProvider"
-        launcher:screen="0"
-        launcher:x="0"
-        launcher:y="1"
-        launcher:spanX="5"
-        launcher:spanY="1" />
-
-    <!-- Google Folder -->
-    <!-- Google, Gmail, Maps, YouTube, Drive, YouTube Music, Play Movies, Duo, Photos -->
-    <folder
-        launcher:title="@string/google_folder_title"
-        launcher:screen="0"
-        launcher:x="0"
-        launcher:y="5">
-        <favorite
-            launcher:packageName="com.google.android.googlequicksearchbox"
-            launcher:className="com.google.android.googlequicksearchbox.SearchActivity"/>
-        <favorite
-            launcher:packageName="com.google.android.gm"
-            launcher:className="com.google.android.gm.ConversationListActivityGmail"/>
-        <favorite
-            launcher:packageName="com.google.android.apps.maps"
-            launcher:className="com.google.android.maps.MapsActivity"/>
-        <favorite
-            launcher:packageName="com.google.android.youtube"
-            launcher:className="com.google.android.youtube.app.honeycomb.Shell$HomeActivity"/>
-        <favorite
-            launcher:packageName="com.google.android.apps.docs"
-            launcher:className="com.google.android.apps.docs.app.NewMainProxyActivity"/>
-        <favorite
-            launcher:packageName="com.google.android.apps.youtube.music"
-            launcher:className="com.google.android.apps.youtube.music.activities.MusicActivity"/>
-        <favorite
-            launcher:packageName="com.google.android.videos"
-            launcher:className="com.google.android.youtube.videos.EntryPoint"/>
-        <favorite
-            launcher:packageName="com.google.android.apps.tachyon"
-            launcher:className="com.google.android.apps.tachyon.MainActivity"/>
-        <favorite
-            launcher:packageName="com.google.android.apps.photos"
-            launcher:className="com.google.android.apps.photos.home.HomeActivity"/>
-    </folder>
-
-    <favorite
-        launcher:screen="0"
-        launcher:x="4"
-        launcher:y="5"
-        launcher:packageName="com.android.vending"
-        launcher:className="com.android.vending.AssetBrowserActivity"/>
-
-    <!-- Screen 1 -->
-    <appwidget
-        launcher:packageName="org.lineageos.twelve"
-        launcher:className="org.lineageos.twelve.ui.widgets.NowPlayingAppWidgetProvider"
-        launcher:screen="1"
-        launcher:x="0"
-        launcher:y="0"
-        launcher:spanX="6"
-        launcher:spanY="1" />
-
-    <resolve
-        launcher:screen="1"
-        launcher:x="0"
-        launcher:y="-1" >
-        <favorite launcher:uri="#Intent;action=android.intent.action.MAIN;category=android.intent.category.APP_EMAIL;end" />
-        <favorite launcher:uri="mailto:" />
-    </resolve>
-
-    <resolve
-        launcher:screen="1"
-        launcher:x="1"
-        launcher:y="-1" >
-        <favorite launcher:uri="#Intent;action=android.intent.action.MAIN;category=android.intent.category.APP_GALLERY;end" />
-        <favorite launcher:uri="#Intent;type=images/*;end" />
-    </resolve>
-
-    <favorite
-        launcher:packageName="org.lineageos.twelve"
-        launcher:className="org.lineageos.twelve.MainActivity"
-        launcher:screen="1"
-        launcher:x="2"
-        launcher:y="-1" />
-
-    <favorite
-        launcher:packageName="com.android.settings"
-        launcher:className="com.android.settings.Settings"
-        launcher:screen="1"
-        launcher:x="3"
-        launcher:y="-1" />
-
-</favorites>
diff --git a/res/xml/device_profiles.xml b/res/xml/device_profiles.xml
index e1527f9cca..aece22cbb7 100644
--- a/res/xml/device_profiles.xml
+++ b/res/xml/device_profiles.xml
@@ -26,32 +26,27 @@
         launcher:numFolderColumns="2"
         launcher:numHotseatIcons="2"
         launcher:dbFile="launcher_2_by_2.db"
+        launcher:isScalable="true"
+        launcher:folderStyle="@style/Folder2x2Style"
         launcher:defaultLayoutId="@xml/default_workspace_2x2"
         launcher:deviceCategory="phone" >
 
         <display-option
-            launcher:name="Super Short Stubby"
-            launcher:minWidthDps="200"
-            launcher:minHeightDps="200"
-            launcher:iconImageSize="120"
-            launcher:iconTextSize="17.0"
-            launcher:allAppsIconSize="60.0"
-            launcher:allAppsIconTextSize="14.0"
             launcher:allAppsBorderSpace="16"
             launcher:allAppsCellHeight="104"
-            launcher:canBeDefault="true" />
-
-        <display-option
-            launcher:name="Shorter Stubby"
-            launcher:minWidthDps="200"
-            launcher:minHeightDps="300"
-            launcher:iconImageSize="120"
-            launcher:iconTextSize="17.0"
-            launcher:allAppsIconSize="60"
-            launcher:allAppsIconTextSize="14.0"
-            launcher:allAppsBorderSpace="16"
-            launcher:allAppsCellHeight="104"
-            launcher:canBeDefault="true" />
+            launcher:allAppsIconSize="66"
+            launcher:allAppsIconTextSize="14"
+            launcher:borderSpace="16"
+            launcher:horizontalMargin="22"
+            launcher:hotseatQsbSpace="36"
+            launcher:iconImageSize="157"
+            launcher:iconSizeLandscape="113"
+            launcher:iconTextSize="24"
+            launcher:minCellHeight="221"
+            launcher:minCellWidth="176"
+            launcher:minHeightDps="838"
+            launcher:minWidthDps="367"
+            launcher:name="Pixel 2021"/>
 
     </grid-option>
 
@@ -64,32 +59,26 @@
         launcher:numFolderColumns="3"
         launcher:numHotseatIcons="3"
         launcher:dbFile="launcher_3_by_3.db"
+        launcher:isScalable="true"
+        launcher:folderStyle="@style/Folder3x3Style"
         launcher:defaultLayoutId="@xml/default_workspace_3x3"
         launcher:deviceCategory="phone" >
 
         <display-option
-            launcher:name="Super Short Stubby"
-            launcher:minWidthDps="255"
-            launcher:minHeightDps="300"
-            launcher:iconImageSize="85"
-            launcher:iconTextSize="15.0"
-            launcher:allAppsIconSize="60"
-            launcher:allAppsIconTextSize="14.0"
-            launcher:allAppsBorderSpace="16"
-            launcher:allAppsCellHeight="104"
-            launcher:canBeDefault="true" />
-
-        <display-option
-            launcher:name="Shorter Stubby"
-            launcher:minWidthDps="255"
-            launcher:minHeightDps="400"
-            launcher:iconImageSize="85"
-            launcher:iconTextSize="15.0"
-            launcher:allAppsIconSize="60"
-            launcher:allAppsIconTextSize="14.0"
             launcher:allAppsBorderSpace="16"
             launcher:allAppsCellHeight="104"
-            launcher:canBeDefault="true" />
+            launcher:allAppsIconSize="66"
+            launcher:allAppsIconTextSize="14"
+            launcher:borderSpace="16"
+            launcher:horizontalMargin="22"
+            launcher:hotseatQsbSpace="36"
+            launcher:iconImageSize="89"
+            launcher:iconTextSize="16"
+            launcher:minCellHeight="165"
+            launcher:minCellWidth="112"
+            launcher:minHeightDps="838"
+            launcher:minWidthDps="367"
+            launcher:name="Pixel 2021"/>
 
     </grid-option>
 
@@ -102,59 +91,24 @@
         launcher:numHotseatIcons="4"
         launcher:numExtendedHotseatIcons="6"
         launcher:dbFile="launcher_4_by_4.db"
+        launcher:isScalable="true"
+        launcher:folderStyle="@style/Folder4x4Style"
         launcher:inlineNavButtonsEndSpacing="@dimen/taskbar_button_margin_split"
         launcher:defaultLayoutId="@xml/default_workspace_4x4"
         launcher:deviceCategory="phone|multi_display" >
 
         <display-option
-            launcher:name="Short Stubby"
-            launcher:minWidthDps="275"
-            launcher:minHeightDps="420"
-            launcher:iconImageSize="48"
-            launcher:iconTextSize="13.0"
-            launcher:allAppsBorderSpace="16"
-            launcher:allAppsCellHeight="104"
-            launcher:canBeDefault="true" />
-
-        <display-option
-            launcher:name="Stubby"
-            launcher:minWidthDps="255"
-            launcher:minHeightDps="450"
-            launcher:iconImageSize="48"
-            launcher:iconTextSize="13.0"
-            launcher:allAppsBorderSpace="16"
-            launcher:allAppsCellHeight="104"
-            launcher:canBeDefault="true" />
-
-        <display-option
-            launcher:name="Nexus S"
-            launcher:minWidthDps="296"
-            launcher:minHeightDps="491.33"
-            launcher:iconImageSize="48"
-            launcher:iconTextSize="13.0"
-            launcher:allAppsBorderSpace="16"
-            launcher:allAppsCellHeight="104"
-            launcher:canBeDefault="true" />
-
-        <display-option
-            launcher:name="Nexus 4"
-            launcher:minWidthDps="359"
-            launcher:minHeightDps="567"
-            launcher:iconImageSize="54"
-            launcher:iconTextSize="13.0"
-            launcher:allAppsBorderSpace="16"
-            launcher:allAppsCellHeight="104"
-            launcher:canBeDefault="true" />
-
-        <display-option
-            launcher:name="Nexus 5"
-            launcher:minWidthDps="335"
-            launcher:minHeightDps="567"
-            launcher:iconImageSize="54"
-            launcher:iconTextSize="13.0"
             launcher:allAppsBorderSpace="16"
             launcher:allAppsCellHeight="104"
-            launcher:canBeDefault="true" />
+            launcher:borderSpace="16"
+            launcher:horizontalMargin="21.5"
+            launcher:iconImageSize="66"
+            launcher:iconTextSize="14"
+            launcher:minCellHeight="130"
+            launcher:minCellWidth="80"
+            launcher:minHeightDps="838"
+            launcher:minWidthDps="367"
+            launcher:name="Pixel 2021"/>
 
     </grid-option>
 
@@ -190,7 +144,7 @@
             launcher:minWidthDps="367"
             launcher:name="Pixel 2021"
             launcher:startAlignTaskbar="true"
-            launcher:startAlignTaskbarLandscape="true" />
+            launcher:startAlignTaskbarLandscape="true"/>
 
     </grid-option>
 
@@ -203,45 +157,28 @@
         launcher:numHotseatIcons="5"
         launcher:numExtendedHotseatIcons="6"
         launcher:dbFile="launcher.db"
+        launcher:isScalable="true"
+        launcher:folderStyle="@style/Folder5x5Style"
         launcher:inlineNavButtonsEndSpacing="@dimen/taskbar_button_margin_split"
         launcher:defaultLayoutId="@xml/default_workspace_5x5"
         launcher:deviceCategory="phone|multi_display" >
 
         <display-option
-            launcher:name="Large Phone"
-            launcher:minWidthDps="406"
-            launcher:minHeightDps="694"
-            launcher:iconImageSize="56"
-            launcher:iconTextSize="12.0"
-            launcher:allAppsBorderSpace="16"
-            launcher:allAppsCellHeight="104"
-            launcher:startAlignTaskbar="true"
-            launcher:startAlignTaskbarLandscape="true"
-            launcher:canBeDefault="true" />
-
-        <display-option
-            launcher:name="Large Phone Split Display"
-            launcher:minWidthDps="406"
-            launcher:minHeightDps="694"
-            launcher:iconImageSize="56"
-            launcher:iconTextSize="12.0"
-            launcher:allAppsBorderSpace="16"
-            launcher:allAppsCellHeight="104"
-            launcher:startAlignTaskbar="true"
-            launcher:startAlignTaskbarLandscape="true"
-            launcher:canBeDefault="true" />
-
-        <display-option
-            launcher:name="Shorter Stubby"
-            launcher:minWidthDps="255"
-            launcher:minHeightDps="400"
-            launcher:iconImageSize="48"
+            launcher:allAppsBorderSpace="16.0"
+            launcher:allAppsCellHeight="104.0"
+            launcher:borderSpace="16.0"
+            launcher:canBeDefault="true"
+            launcher:horizontalMargin="22.0"
+            launcher:iconImageSize="57.0"
+            launcher:iconSizeLandscape="59.0"
             launcher:iconTextSize="12.0"
-            launcher:allAppsBorderSpace="16"
-            launcher:allAppsCellHeight="104"
+            launcher:minCellHeight="108.0"
+            launcher:minCellWidth="61.0"
+            launcher:minHeightDps="750.0"
+            launcher:minWidthDps="387.0"
+            launcher:name="Pixel 2020"
             launcher:startAlignTaskbar="true"
-            launcher:startAlignTaskbarLandscape="true"
-            launcher:canBeDefault="true" />
+            launcher:startAlignTaskbarLandscape="true"/>
 
     </grid-option>
 
@@ -253,21 +190,28 @@
         launcher:numFolderColumns="4"
         launcher:numHotseatIcons="5"
         launcher:dbFile="launcher_5_by_6.db"
+        launcher:isScalable="true"
+        launcher:folderStyle="@style/Folder5x6Style"
+        launcher:inlineNavButtonsEndSpacing="@dimen/taskbar_button_margin_split"
         launcher:defaultLayoutId="@xml/default_workspace_5x6"
         launcher:deviceCategory="phone|multi_display" >
 
         <display-option
-            launcher:name="Large Phone"
-            launcher:minWidthDps="406"
-            launcher:minHeightDps="694"
-            launcher:iconImageSize="56"
+            launcher:allAppsBorderSpace="16.0"
+            launcher:allAppsCellHeight="104.0"
+            launcher:borderSpace="16.0"
+            launcher:canBeDefault="true"
+            launcher:horizontalMargin="22.0"
+            launcher:iconImageSize="57.0"
+            launcher:iconSizeLandscape="59.0"
             launcher:iconTextSize="12.0"
-            launcher:allAppsBorderSpace="16"
-            launcher:allAppsCellHeight="104"
+            launcher:minCellHeight="90.0"
+            launcher:minCellWidth="61.0"
+            launcher:minHeightDps="750.0"
+            launcher:minWidthDps="387.0"
+            launcher:name="Pixel 2020"
             launcher:startAlignTaskbar="true"
-            launcher:startAlignTaskbarLandscape="true"
-            launcher:canBeDefault="true" />
-
+            launcher:startAlignTaskbarLandscape="true"/>
     </grid-option>
 
     <grid-option
@@ -278,183 +222,222 @@
         launcher:numFolderColumns="4"
         launcher:numHotseatIcons="5"
         launcher:dbFile="launcher_5_by_7.db"
+        launcher:isScalable="true"
+        launcher:folderStyle="@style/Folder5x7Style"
+        launcher:inlineNavButtonsEndSpacing="@dimen/taskbar_button_margin_split"
         launcher:defaultLayoutId="@xml/default_workspace_5x7"
         launcher:deviceCategory="phone|multi_display" >
 
         <display-option
-            launcher:name="Large Phone"
-            launcher:minWidthDps="406"
-            launcher:minHeightDps="694"
-            launcher:iconImageSize="56"
+            launcher:allAppsBorderSpace="16.0"
+            launcher:allAppsCellHeight="104.0"
+            launcher:borderSpace="16.0"
+            launcher:canBeDefault="true"
+            launcher:horizontalMargin="22.0"
+            launcher:iconImageSize="57.0"
+            launcher:iconSizeLandscape="59.0"
             launcher:iconTextSize="12.0"
-            launcher:allAppsBorderSpace="16"
-            launcher:allAppsCellHeight="104"
+            launcher:minCellHeight="77.14"
+            launcher:minCellWidth="61.0"
+            launcher:minHeightDps="750.0"
+            launcher:minWidthDps="387.0"
+            launcher:name="Pixel 2020"
             launcher:startAlignTaskbar="true"
-            launcher:startAlignTaskbarLandscape="true"
-            launcher:canBeDefault="true" />
+            launcher:startAlignTaskbarLandscape="true"/>
 
     </grid-option>
 
     <grid-option
-        launcher:name="6_by_5"
-        launcher:numRows="5"
+        launcher:allAppsStyle="@style/AllAppsStyle6x5"
+        launcher:cellStyle="@style/CellStyle6x5"
+        launcher:dbFile="launcher_6_by_5.db"
+        launcher:defaultLayoutId="@xml/default_workspace_6x5"
+        launcher:deviceCategory="tablet"
+        launcher:devicePaddingId="@xml/paddings_6x5"
+        launcher:folderStyle="@style/Folder6x5Style"
+        launcher:inlineNavButtonsEndSpacing="@dimen/taskbar_button_margin_6_5"
+        launcher:inlineQsb="landscape"
+        launcher:isScalable="true"
+        launcher:name="tablet_normal"
+        launcher:numAllAppsColumns="6"
         launcher:numColumns="6"
-        launcher:numSearchContainerColumns="3"
-        launcher:numFolderRows="3"
         launcher:numFolderColumns="3"
+        launcher:numFolderRows="3"
         launcher:numHotseatIcons="6"
-        launcher:numAllAppsColumns="6"
-        launcher:isScalable="true"
-        launcher:inlineNavButtonsEndSpacing="@dimen/taskbar_button_margin_6_5"
-        launcher:devicePaddingId="@xml/paddings_6x5"
-        launcher:dbFile="launcher_6_by_5.db"
-        launcher:defaultLayoutId="@xml/default_workspace_6x5"
-        launcher:deviceCategory="tablet" >
+        launcher:numRows="5"
+        launcher:numSearchContainerColumns="3">
 
         <display-option
-            launcher:name="Tablet"
-            launcher:minWidthDps="900"
-            launcher:minHeightDps="820"
-            launcher:minCellHeight="120"
-            launcher:minCellWidth="102"
-            launcher:minCellHeightLandscape="104"
-            launcher:minCellWidthLandscape="120"
-            launcher:iconImageSize="60"
-            launcher:iconTextSize="14"
+            launcher:allAppsBorderSpaceHorizontal="8"
+            launcher:allAppsBorderSpaceLandscape="16"
+            launcher:allAppsBorderSpaceVertical="16"
+            launcher:allAppsCellHeight="142"
+            launcher:allAppsCellHeightLandscape="126"
+            launcher:allAppsCellWidth="96"
+            launcher:allAppsCellWidthLandscape="126"
+            launcher:allAppsIconSize="66"
+            launcher:allAppsIconTextSize="14"
             launcher:borderSpaceHorizontal="16"
-            launcher:borderSpaceVertical="64"
             launcher:borderSpaceLandscapeHorizontal="64"
             launcher:borderSpaceLandscapeVertical="16"
+            launcher:borderSpaceVertical="64"
+            launcher:canBeDefault="true"
             launcher:horizontalMargin="54"
             launcher:horizontalMarginLandscape="120"
-            launcher:allAppsCellWidth="96"
+            launcher:hotseatBarBottomSpace="36"
+            launcher:hotseatBarBottomSpaceLandscape="40"
+            launcher:hotseatQsbSpace="32"
+            launcher:iconImageSize="66"
+            launcher:iconTextSize="14"
+            launcher:minCellHeight="120"
+            launcher:minCellHeightLandscape="104"
+            launcher:minCellWidth="102"
+            launcher:minCellWidthLandscape="120"
+            launcher:minHeightDps="748"
+            launcher:minWidthDps="800"
+            launcher:name="Pixel C"
+            launcher:startAlignTaskbar="true"
+            launcher:startAlignTaskbarLandscape="true"
+            launcher:transientTaskbarIconSize="57.2"/>
+
+        <display-option
+            launcher:allAppsBorderSpaceHorizontal="8"
+            launcher:allAppsBorderSpaceLandscapeHorizontal="40"
+            launcher:allAppsBorderSpaceLandscapeVertical="16"
+            launcher:allAppsBorderSpaceVertical="20"
             launcher:allAppsCellHeight="142"
+            launcher:allAppsCellHeightLandscape="150"
+            launcher:allAppsCellWidth="120"
             launcher:allAppsCellWidthLandscape="126"
-            launcher:allAppsCellHeightLandscape="126"
-            launcher:allAppsIconSize="60"
+            launcher:allAppsIconSize="66"
             launcher:allAppsIconTextSize="14"
-            launcher:allAppsBorderSpaceHorizontal="8"
-            launcher:allAppsBorderSpaceVertical="16"
-            launcher:allAppsBorderSpaceLandscape="16"
-            launcher:hotseatBarBottomSpace="76"
+            launcher:borderSpaceHorizontal="16"
+            launcher:borderSpaceLandscapeHorizontal="90"
+            launcher:borderSpaceLandscapeVertical="16"
+            launcher:borderSpaceVertical="110"
+            launcher:canBeDefault="true"
+            launcher:horizontalMargin="70"
+            launcher:horizontalMarginLandscape="169"
+            launcher:hotseatBarBottomSpace="36"
             launcher:hotseatBarBottomSpaceLandscape="40"
+            launcher:hotseatQsbSpace="32"
+            launcher:iconImageSize="66"
+            launcher:iconTextSize="14"
+            launcher:minCellHeight="120"
+            launcher:minCellHeightLandscape="130"
+            launcher:minCellWidth="120"
+            launcher:minCellWidthLandscape="120"
+            launcher:minHeightDps="880"
+            launcher:minWidthDps="941"
+            launcher:name="Pixel C (display: small)"
             launcher:startAlignTaskbar="true"
             launcher:startAlignTaskbarLandscape="true"
-            launcher:canBeDefault="true" />
-
-    </grid-option>
-
-    <grid-option
-        launcher:name="desktop_6_by_5"
-        launcher:numRows="5"
-        launcher:numColumns="6"
-        launcher:numSearchContainerColumns="3"
-        launcher:numFolderRows="3"
-        launcher:numFolderColumns="3"
-        launcher:numHotseatIcons="6"
-        launcher:numAllAppsColumns="6"
-        launcher:isScalable="true"
-        launcher:inlineNavButtonsEndSpacing="@dimen/taskbar_button_margin_6_5"
-        launcher:devicePaddingId="@xml/paddings_6x5"
-        launcher:dbFile="launcher_6_by_5.db"
-        launcher:defaultLayoutId="@xml/default_workspace_6x5"
-        launcher:deviceCategory="desktop" >
+            launcher:transientTaskbarIconSize="57.2"/>
 
         <display-option
-            launcher:name="Desktop"
-            launcher:minWidthDps="900"
-            launcher:minHeightDps="820"
-            launcher:minCellHeight="120"
-            launcher:minCellWidth="102"
-            launcher:minCellHeightLandscape="104"
-            launcher:minCellWidthLandscape="120"
-            launcher:iconImageSize="60"
-            launcher:iconTextSize="14"
-            launcher:borderSpaceHorizontal="16"
-            launcher:borderSpaceVertical="64"
-            launcher:borderSpaceLandscapeHorizontal="64"
-            launcher:borderSpaceLandscapeVertical="16"
-            launcher:horizontalMargin="54"
-            launcher:horizontalMarginLandscape="120"
-            launcher:allAppsCellWidth="96"
+            launcher:allAppsBorderSpaceHorizontal="8"
+            launcher:allAppsBorderSpaceLandscapeHorizontal="12"
+            launcher:allAppsBorderSpaceLandscapeVertical="16"
+            launcher:allAppsBorderSpaceVertical="12"
             launcher:allAppsCellHeight="142"
+            launcher:allAppsCellHeightLandscape="110"
+            launcher:allAppsCellWidth="92"
             launcher:allAppsCellWidthLandscape="126"
-            launcher:allAppsCellHeightLandscape="126"
-            launcher:allAppsIconSize="60"
+            launcher:allAppsIconSize="66"
             launcher:allAppsIconTextSize="14"
-            launcher:allAppsBorderSpaceHorizontal="8"
-            launcher:allAppsBorderSpaceVertical="16"
-            launcher:allAppsBorderSpaceLandscape="16"
-            launcher:hotseatBarBottomSpace="76"
+            launcher:borderSpaceHorizontal="16"
+            launcher:borderSpaceLandscapeHorizontal="50"
+            launcher:borderSpaceLandscapeVertical="16"
+            launcher:borderSpaceVertical="39.5"
+            launcher:canBeDefault="true"
+            launcher:horizontalMargin="45"
+            launcher:horizontalMarginLandscape="94"
+            launcher:hotseatBarBottomSpace="36"
             launcher:hotseatBarBottomSpaceLandscape="40"
-            launcher:canBeDefault="true" />
-
-    </grid-option>
-
-    <grid-option
-        launcher:name="6_by_6"
-        launcher:numRows="6"
-        launcher:numColumns="6"
-        launcher:numFolderRows="5"
-        launcher:numFolderColumns="5"
-        launcher:numHotseatIcons="6"
-        launcher:dbFile="launcher_6_by_6.db"
-        launcher:defaultLayoutId="@xml/default_workspace_6x6"
-        launcher:deviceCategory="phone|multi_display" >
-
-        <display-option
-            launcher:name="Large Phone"
-            launcher:minWidthDps="406"
-            launcher:minHeightDps="694"
-            launcher:iconImageSize="48"
-            launcher:iconTextSize="12.0"
-            launcher:allAppsBorderSpace="16"
-            launcher:allAppsCellHeight="104"
+            launcher:hotseatQsbSpace="32"
+            launcher:iconImageSize="66"
+            launcher:iconTextSize="14"
+            launcher:minCellHeight="120"
+            launcher:minCellHeightLandscape="90"
+            launcher:minCellWidth="92"
+            launcher:minCellWidthLandscape="120"
+            launcher:minHeightDps="676"
+            launcher:minWidthDps="723"
+            launcher:name="Pixel C (display: large)"
             launcher:startAlignTaskbar="true"
             launcher:startAlignTaskbarLandscape="true"
-            launcher:canBeDefault="true" />
+            launcher:transientTaskbarIconSize="57.2"/>
 
-    </grid-option>
-
-    <grid-option
-        launcher:name="fixed_landscape_mode"
-        launcher:numFolderRows="2"
-        launcher:numFolderColumns="3"
-        launcher:numFolderRowsLandscape="2"
-        launcher:folderStyle="@style/FolderStyleDefault"
-        launcher:numHotseatIcons="4"
-        launcher:numExtendedHotseatIcons="8"
-        launcher:numAllAppsColumns="8"
-        launcher:numExtendedAllAppsColumns="8"
-        launcher:workspaceSpecsId="@xml/spec_handheld_workspace_3_row"
-        launcher:allAppsSpecsId="@xml/spec_handheld_all_apps_3_row"
-        launcher:folderSpecsId="@xml/spec_handheld_folder_3_row"
-        launcher:hotseatSpecsId="@xml/spec_handheld_hotseat_3_row"
-        launcher:workspaceCellSpecsId="@xml/spec_handheld_workspace_cell_3_row"
-        launcher:allAppsCellSpecsId="@xml/spec_all_apps_cell_match_workspace"
-        launcher:isScalable="true"
-        launcher:inlineQsb="landscape"
-        launcher:devicePaddingId="@xml/paddings_3_row"
-        launcher:gridSizeSpecsId="@xml/spec_col_count_3_row"
-        launcher:isFixedLandscape="true"
-        launcher:defaultLayoutId="@xml/default_workspace_8x3"
-        launcher:deviceCategory="phone">
+        <display-option
+            launcher:allAppsBorderSpaceHorizontal="8"
+            launcher:allAppsBorderSpaceLandscapeHorizontal="8"
+            launcher:allAppsBorderSpaceLandscapeVertical="16"
+            launcher:allAppsBorderSpaceVertical="8"
+            launcher:allAppsCellHeight="142"
+            launcher:allAppsCellHeightLandscape="94"
+            launcher:allAppsCellWidth="84"
+            launcher:allAppsCellWidthLandscape="126"
+            launcher:allAppsIconSize="66"
+            launcher:allAppsIconSizeLandscape="52"
+            launcher:allAppsIconTextSize="14"
+            launcher:borderSpaceHorizontal="16"
+            launcher:borderSpaceLandscapeHorizontal="38"
+            launcher:borderSpaceLandscapeVertical="16"
+            launcher:borderSpaceVertical="16"
+            launcher:canBeDefault="true"
+            launcher:horizontalMargin="36"
+            launcher:horizontalMarginLandscape="71"
+            launcher:hotseatBarBottomSpace="36"
+            launcher:hotseatBarBottomSpaceLandscape="40"
+            launcher:hotseatQsbSpace="32"
+            launcher:iconImageSize="66"
+            launcher:iconSizeLandscape="52"
+            launcher:iconTextSize="14"
+            launcher:minCellHeight="120"
+            launcher:minCellHeightLandscape="79.5"
+            launcher:minCellWidth="84"
+            launcher:minCellWidthLandscape="120"
+            launcher:minHeightDps="613"
+            launcher:minWidthDps="656"
+            launcher:name="Pixel C (display: larger)"
+            launcher:startAlignTaskbar="true"
+            launcher:startAlignTaskbarLandscape="true"/>
 
         <display-option
-            launcher:name="Fixed Landscape"
-            launcher:minWidthDps="387"
-            launcher:minHeightDps="750"
-            launcher:minCellHeight="108"
-            launcher:minCellWidth="61"
-            launcher:borderSpace="16"
-            launcher:horizontalMargin="22"
-            launcher:iconImageSize="57"
-            launcher:iconSizeLandscape="59"
-            launcher:iconTextSize="12"
-            launcher:allAppsBorderSpace="16"
-            launcher:allAppsCellHeight="104"
-            launcher:canBeDefault="true" />
+            launcher:allAppsBorderSpaceHorizontal="8"
+            launcher:allAppsBorderSpaceLandscapeHorizontal="8"
+            launcher:allAppsBorderSpaceLandscapeVertical="16"
+            launcher:allAppsBorderSpaceVertical="4"
+            launcher:allAppsCellHeight="142"
+            launcher:allAppsCellHeightLandscape="84"
+            launcher:allAppsCellWidth="78"
+            launcher:allAppsCellWidthLandscape="126"
+            launcher:allAppsIconSize="66"
+            launcher:allAppsIconSizeLandscape="48"
+            launcher:allAppsIconTextSize="14"
+            launcher:borderSpaceHorizontal="16"
+            launcher:borderSpaceLandscapeHorizontal="28"
+            launcher:borderSpaceLandscapeVertical="16"
+            launcher:borderSpaceVertical="16"
+            launcher:canBeDefault="true"
+            launcher:horizontalMargin="31"
+            launcher:horizontalMarginLandscape="53"
+            launcher:hotseatBarBottomSpace="36"
+            launcher:hotseatBarBottomSpaceLandscape="40"
+            launcher:hotseatQsbSpace="32"
+            launcher:iconImageSize="66"
+            launcher:iconSizeLandscape="48"
+            launcher:iconTextSize="14"
+            launcher:minCellHeight="120"
+            launcher:minCellHeightLandscape="74"
+            launcher:minCellWidth="77"
+            launcher:minCellWidthLandscape="120"
+            launcher:minHeightDps="564"
+            launcher:minWidthDps="603"
+            launcher:name="Pixel C (display: largest)"
+            launcher:startAlignTaskbar="true"
+            launcher:startAlignTaskbarLandscape="true"/>
 
     </grid-option>
 
-</profiles>
\ No newline at end of file
+</profiles>
diff --git a/src/com/android/launcher3/LauncherFiles.java b/src/com/android/launcher3/LauncherFiles.java
index 5bb6f6852f..c266774ba9 100644
--- a/src/com/android/launcher3/LauncherFiles.java
+++ b/src/com/android/launcher3/LauncherFiles.java
@@ -18,7 +18,6 @@ public class LauncherFiles {
     public static final String LAUNCHER_DB = "launcher.db";
     public static final String LAUNCHER_5_BY_8_DB = "launcher_5_by_8.db";
     public static final String LAUNCHER_5_BY_7_DB = "launcher_5_by_7.db";
-    public static final String LAUNCHER_6_BY_6_DB = "launcher_6_by_6.db";
     public static final String LAUNCHER_6_BY_5_DB = "launcher_6_by_5.db";
     public static final String LAUNCHER_4_BY_5_DB = "launcher_4_by_5.db";
     public static final String LAUNCHER_4_BY_6_DB = "launcher_4_by_6.db";
@@ -48,7 +47,6 @@ public class LauncherFiles {
             LAUNCHER_DB,
             LAUNCHER_5_BY_8_DB,
             LAUNCHER_5_BY_7_DB,
-            LAUNCHER_6_BY_6_DB,
             LAUNCHER_6_BY_5_DB,
             LAUNCHER_4_BY_5_DB,
             LAUNCHER_4_BY_6_DB,
-- 
2.56.0

